File: src/main/java/krasa/editorGroups/actions/ToggleHideEmptyAction.java
package krasa.editorGroups.actions;

import com.intellij.openapi.actionSystem.ActionUpdateThread;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.ToggleAction;
import com.intellij.openapi.project.DumbAware;
import krasa.editorGroups.EditorGroupsSettingsState;
import org.jetbrains.annotations.NotNull;

public class ToggleHideEmptyAction extends ToggleAction implements DumbAware {
  @Override
  public @NotNull ActionUpdateThread getActionUpdateThread() {
    return ActionUpdateThread.BGT;
  }

  @Override
  public boolean isSelected(@NotNull AnActionEvent e) {
    return EditorGroupsSettingsState.state().isHideEmpty();
  }

  @Override
  public void setSelected(@NotNull AnActionEvent e, boolean state) {
    EditorGroupsSettingsState.state().setHideEmpty(state);
  }
}

File: src/main/java/krasa/editorGroups/actions/ToggleHideEmptyAction.kt
package krasa.editorGroups.actions

import com.intellij.openapi.actionSystem.ActionUpdateThread
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.ToggleAction
import com.intellij.openapi.project.DumbAware
import krasa.editorGroups.messages.EditorGroupsBundle.message
import krasa.editorGroups.settings.EditorGroupsSettings
import krasa.editorGroups.support.Notifications

class ToggleHideEmptyAction : ToggleAction(), DumbAware {
  override fun getActionUpdateThread(): ActionUpdateThread = ActionUpdateThread.BGT

  override fun isSelected(e: AnActionEvent): Boolean = EditorGroupsSettings.instance.isHideEmpty

  override fun setSelected(e: AnActionEvent, state: Boolean) {
    EditorGroupsSettings.instance.isHideEmpty = state
    EditorGroupsSettings.instance.fireChanged()
    Notifications.notifyState(message("hide.empty.groups"), state)
  }

  companion object {
    const val ID: String = "krasa.editorGroups.ToggleHideEmpty"
  }
}
